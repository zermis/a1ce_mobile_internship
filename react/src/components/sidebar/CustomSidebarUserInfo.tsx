import React from 'react';
import { View, Text, Image, StyleSheet } from 'react-native';

import Colors from '../../constants/Colors';

interface Props {
    userName: string;
    userEmail: string;
    userStudentID: string;
    userProfile: string;
}

const CustomSidebarUserInfo = (props: Props) => {
    const image = (
        <Image
            style={styles.userPicture}
            source={
                props.userProfile
                    ? { uri: props.userProfile }
                    : require('../../assets/images/profile.png')
            }
        />
    );
    const userName = props.userName ? (
        <Text style={styles.userName}>{props.userName}</Text>
    ) : null;
    const userEmail = props.userEmail ? (
        <Text style={styles.userEmail}>{props.userEmail}</Text>
    ) : null;
    const userStudentID = props.userStudentID ? (
        <Text style={styles.userStudentID}>{props.userStudentID}</Text>
    ) : null;

    return (
        <View style={styles.userInfo}>
            {image}
            {userName}
            {userStudentID}
            {userEmail}
        </View>
    );
};

const styles = StyleSheet.create({
    userInfo: {
        justifyContent: 'center',
        marginVertical: 38,
        alignItems: 'center'
    },
    userPicture: {
        height: 80,
        width: 80,
        borderRadius: 80 / 2,
        backgroundColor: '#222'
    },
    userName: {
        fontSize: 25,
        fontFamily: 'ibm-bold',
        color: Colors.nameColor,
        marginTop: 9,
        marginBottom: 2.5,
        textAlign: 'center'
    },
    userStudentID: {
        fontSize: 17,
        fontFamily: 'ibm-regular',
        color: Colors.userInfoColor,
        marginLeft: '1%',
        marginBottom: 2.5
    },
    userEmail: {
        fontSize: 17,
        fontFamily: 'ibm-regular',
        color: Colors.userInfoColor,
        marginLeft: '1%',
        marginBottom: -8
    }
});

export default CustomSidebarUserInfo;
