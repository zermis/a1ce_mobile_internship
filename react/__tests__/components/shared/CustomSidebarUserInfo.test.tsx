import CustomSidebarUserInfo from '../../../src/components/sidebar/CustomSidebarUserInfo';
import renderer from 'react-test-renderer';

it('renders custom sidebar user info correctly', function () {
    var tree = renderer
        .create(
            <CustomSidebarUserInfo
                userName={''}
                userEmail={''}
                userStudentID={''}
                userProfile={''}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
