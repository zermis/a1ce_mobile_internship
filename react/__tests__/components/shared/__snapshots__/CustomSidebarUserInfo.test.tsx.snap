// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom sidebar user info correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginVertical": 38,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "backgroundColor": "#222",
        "borderRadius": 40,
        "height": 80,
        "width": 80,
      }
    }
  />
</View>
`;
